### PR TITLE
v2

### PR DESCRIPTION
**Pull request by adam-gligor**

### CODE REVIEW

File: v2/readme.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+vehicle has made over time. 
+
+There are many different use-cases for this, one of them is taxation. 
+Our customers e.g. maintain a logbook, to keep track of how many trips were 
+made privately and how many kilometers were driven in a business context.
+
+
+## The trip extraction
+
+The central pice of the logbook is the trip extraction, the process of detecting movement
```
Comment (CashWasabi) [inline v2/readme.md]:
```suggestion
The central piece of the logbook is the trip extraction, the process of detecting movement 
```

File: v2/readme.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+There are many different use-cases for this, one of them is taxation. 
+Our customers e.g. maintain a logbook, to keep track of how many trips were 
+made privately and how many kilometers were driven in a business context.
+
+
+## The trip extraction
+
+The central pice of the logbook is the trip extraction, the process of detecting movement 
+in the gps data and determinging when a vehicle starts and stops moving.
```
Comment (CashWasabi) [inline v2/readme.md]:
```suggestion
in the gps data and determining when a vehicle starts and stops moving.
```

File: v2/readme.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+
+When driving from A to B, a user will probably have interruptions like stopping at red lights, 
+traffic jams, tunels.  The expected behaviour is to have one trip accounting for 
+the whole journey, and not broken trips like from A to first red light etc.
+
+
+GPS technology has some limitations which the trip extraction should account for. 
+Within cities, the accuracy of GPS can be below 15 meters and the GPS position might even "jump", 
+due to bad reception caused by reflections on building. There are places where is not gps signal
```
Comment (CashWasabi) [inline v2/readme.md]:
```suggestion
due to bad reception caused by reflections on building. There are places whithout gps signal
```

File: v2/readme.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+like tunels and undregraound parkings.
```
Comment (CashWasabi) [inline v2/readme.md]:
```suggestion
like tunnels and underground parking lots.
```

File: v2/readme.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+
+## The task
+
+Your challange is to write a simplified version of the trip extraction program.
```
Comment (CashWasabi) [inline v2/readme.md]:
```suggestion
Your challenge is to write a simplified version of the trip extraction.
```

File: v2/readme.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+- The trip start conditions is: a stopped vehicle leaves a circle of 20 meters diameter 
+around it's rest position.
+- The trip end conditions is: a moving vehicle remains within a circle of 20 meters 
+diameter for longer than 5 minutes.
+- Small distances of less than 10 meters between waypoints should be ignored.
+- Large distances in a short time, more than 10 km/minute between waypoints should be also ignored.
+
+
+The input is supplied in a file and contains of a list of Waypoints, this is the gps information
```
Comment (CashWasabi) [inline v2/readme.md]:
```suggestion
The input is supplied in a file and contains a list of Waypoints, this is the gps information 
```

File: v2/readme.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+
+## The trip extraction
+
+The central pice of the logbook is the trip extraction, the process of detecting movement 
+in the gps data and determinging when a vehicle starts and stops moving.
+
+
+When driving from A to B, a user will probably have interruptions like stopping at red lights, 
+traffic jams, tunels.  The expected behaviour is to have one trip accounting for
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
traffic jams, tunnels.  The expected behaviour is to have one trip accounting for 
```

File: v2/readme.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+in the gps data and determinging when a vehicle starts and stops moving.
+
+
+When driving from A to B, a user will probably have interruptions like stopping at red lights, 
+traffic jams, tunels.  The expected behaviour is to have one trip accounting for 
+the whole journey, and not broken trips like from A to first red light etc.
+
+
+GPS technology has some limitations which the trip extraction should account for. 
+Within cities, the accuracy of GPS can be below 15 meters and the GPS position might even "jump",
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
Within cities, location precision may be poorer than 15 meters and the GPS position might even "jump", 
```

File: v2/readme.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+
+## The task
+
+Your challange is to write a simplified version of the trip extraction program.
+
+
+For the trip extraction use the following logic:
+
+- The trip start conditions is: a stopped vehicle leaves a circle of 20 meters diameter
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
- The trip start condition is: a stopped vehicle leaves a circle of 20 meters diameter 
```

File: v2/readme.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+## The task
+
+Your challange is to write a simplified version of the trip extraction program.
+
+
+For the trip extraction use the following logic:
+
+- The trip start conditions is: a stopped vehicle leaves a circle of 20 meters diameter 
+around it's rest position.
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
around its' rest position.
```

File: v2/readme.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+- The trip end conditions is: a moving vehicle remains within a circle of 20 meters
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
- The trip end condition is: a moving vehicle remains within a circle of 20 meters 
```

File: v2/readme.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+- `end` – object, the end of the trip in Waypoint schema (see above)
+- `distance` – int, the distance of the trip in meters
+
+The distance is the the sum of all distances between the individual waypoints of the trip and has to be 
+calculated.
+
+
+## Evaluation
+
+The focus of the evaluation will be clean, readable and tested code, which follows best practises.
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
The focus of the evaluation will be clean, readable and tested code, which follows best practices.
```

File: v2/readme.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+- The trip start conditions is: a stopped vehicle leaves a circle of 20 meters diameter 
+around it's rest position.
+- The trip end conditions is: a moving vehicle remains within a circle of 20 meters 
+diameter for longer than 5 minutes.
+- Small distances of less than 10 meters between waypoints should be ignored.
+- Large distances in a short time, more than 10 km/minute between waypoints should be also ignored.
+
+
+The input is supplied in a file and contains of a list of Waypoints, this is the gps information 
+collected for one vehicle. A waypoints has the following structure.
```
Comment (ppbdrinker) [inline v2/readme.md]:
```suggestion
collected for one vehicle. Waypoints have the following structure:
```

**Comment by adam-gligor** [comment]
tnx for finding the typing mistakes. I didn't have a spellchecker in vs-code, now I have one :)

**Review by CashWasabi — APPROVED** [review]

**Review by ppbdrinker — APPROVED** [review]

**Comment by adam-gligor** [comment]
moved to https://github.com/vimcar/backend-challenge-trip-extraction-2021

**Pull request closed without merging**